toolchain/check/import_cpp.cpp

```diff
@@ -1013,6 +1028,12 @@ static auto ImportFunctionDecl(Context& context, SemIR::LocId loc_id,
        .clang_decl_id = context.sem_ir().clang_decls().Add(
            {.decl = clang_decl, .inst_id = decl_id})}};
 
+  if (IsCppThunkRequired(context, function_info)) {
+    clang::FunctionDecl* thunk_decl = BuildCppThunk(context, function_info);
+    function_info.SetCppThunkRequired(
+        ImportFunctionDecl(context, loc_id, thunk_decl));
```

Comment:
I expect clang-tidy will complain about the recursion here. Maybe we could factor out the body of `ImportFunctionDecl`, and call it twice (once to import the requested function and once to import the thunk if needed) to avoid the appearance of unbounded recursion.